src/os/bluestore/BlueStore.cc

```diff
@@ -3300,8 +3304,27 @@ void BlueStore::handle_conf_change(const struct md_config_t *conf,
 
 void BlueStore::_set_compression()
 {
-  comp_min_blob_size = cct->_conf->bluestore_compression_min_blob_size;
-  comp_max_blob_size = cct->_conf->bluestore_compression_max_blob_size;
+  if (cct->_conf->bluestore_compression_max_blob_size) {
+    comp_min_blob_size = cct->_conf->bluestore_compression_max_blob_size;
+  } else {
+    assert(bdev);
+    if (bdev->is_rotational()) {
```

Comment:
need to update the caller in handle_conf_chagne with if (bdev) guard, and then call this again after bdev is initialized in mkfs() and _open_super_meta() (match _set_alloc_sizes() callers)